frontend/src/components/question/modals/QuestionDetailsModal.tsx

```diff
@@ -26,6 +27,8 @@ interface Props {
 
 const QuestionDetailsModal: React.FC<Props> =
   ({ isVisible, data, closeHandler, editModalHandler: editHandler, deleteHandler }) => {
+    const userRole = LocalStorageHandler.getUserData()!.role;
```

Comment:
I think this crashes the site if a user accesses /home without being logged in

Can try something like
```
    const userData = LocalStorageHandler.getUserData();
    const userRole = userData? userData.role : null;
````
as a short-term fix, but it shows the question page for a split second before redirecting to the login page

Reply:
don't think there is a way to prevent the split second because a request has to be sent to the backend for authentication before loading the page so I just added a loading screen